searchcore/src/vespa/searchcore/proton/server/proton_configurer.cpp

```diff
@@ -54,13 +54,30 @@ ProtonConfigurer::ProtonConfigurer(vespalib::SyncableThreadExecutor &executor,
       _mutex(),
       _allowReconfig(false),
       _componentConfig(),
-      _diskLayout(diskLayout)
+      _diskLayout(diskLayout),
+      _pendingReconfigureTasks(0)
 {
 }
 
-ProtonConfigurer::~ProtonConfigurer()
-{
-}
+class ProtonConfigurer::ReconfigureTask : public vespalib::Executor::Task {
```

Comment:
Why not use MonitoredRefCount and RetainGuard ?

Reply:
Fallen off the stack. Fixed.